File: sdk/java/io.stockx.sdk.examples/src/main/java/io/stockx/sdk/examples/base/LoginUtil.java
package io.stockx.sdk.examples.base;

import java.util.List;

import io.stockx.client.ApiException;
import io.stockx.client.ApiResponse;
import io.stockx.client.api.StockXApi;
import io.stockx.client.model.CustomerObject;
import io.stockx.client.model.LoginRequest;
import io.stockx.client.model.LoginResponse;

public class LoginUtil {

	public static CustomerObject login(StockXApi api, String username, String password) throws ApiException {
		LoginRequest loginReq = new LoginRequest()
				.email(username)
				.password(password);

		try {
			ApiResponse<LoginResponse> loginWithHttpInfo = api.loginWithHttpInfo(loginReq);
			List<String> list = loginWithHttpInfo.getHeaders().get(ExampleConstants.JWT_HEADER);
			
			String authToken = list.get(0);
			api.getApiClient().addDefaultHeader("jwt-authorization", authToken);
			
			return loginWithHttpInfo.getData().getCustomer();
		} catch (ApiException e) {
			throw e;
		}
	}
	
}

File: sdk/php/StockX-SDK-PHP/src/LoginUtil.php
<?php

class LoginUtil {
    
    public static function login($api, $headerSelector, $username, $password) {
        $login = new \Swagger\Client\Model\LoginRequest();
        $login->setEmail($username);
        $login->setPassword($password);
        
        try {
            $result = $api->loginWithHttpInfo($login);
            $jwtHeader = $result[2]["Jwt-Authorization"][0];
            $headerSelector->setToken($jwtHeader);
            
            return $result;
        } catch (Exception $e) {
            echo 'Exception when calling StockXApi->login: ', $e->getMessage(), PHP_EOL;
        }
    }
}

